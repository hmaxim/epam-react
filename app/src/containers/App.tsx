import * as React from 'react';

export class App extends React.Component<any, {}> {
    render() {
        return <h1>Hello from { this.props.title }!</h1>;
    }
}