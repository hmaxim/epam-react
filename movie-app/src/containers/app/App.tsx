import React, { useCallback, useEffect } from 'react';
import { connect } from 'react-redux';
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch,
  Link,
} from 'react-router-dom';
import { IMovie } from '../../interfaces/IMovie';
import {
  getMovieById,
  loadMovies,
  setSearchParams,
} from '../../redux/rootActions';
import EmptyState from '../../shared-components/empty-state/EmptyState';
import MovieItem from '../../shared-components/movie-item/MovieItem';
import ErrorBoundary from '../error-boundary/ErrorBoundary';
import HeaderContainer from '../header-container/HeaderContainer';
import ListHeaderContainer from '../list-header-container/ListHeaderContainer';
import MovieDetailsContainer from '../movie-details-container/MovieDetailsContainer';
import MoviesListWrapper from '../movies-list-container/MoviesListWrapper';
import SearchContainer from '../search-container/SearchContainer';
import AppWrapper from './AppWrapper';
import { useHistory, useLocation } from 'react-router-dom';
import { useQuery } from '../../constants/constants';

const App = (props: any) => {
  const history = useHistory();
  const location = useLocation();
  const query = useQuery(location.search);

  useEffect(() => {
    if (!history.location.pathname.includes('film')) {
      if (!props.movies.length) {
        props.setSearchParams({
          searchBy: query.get('searchBy'),
          search: query.get('search'),
          sortBy: query.get('sortBy'),
        });
      }
      props.loadMovies();
    }
  }, [location]);

  const navigate = useCallback((id: number) => {
    history.push(`/film/${id}`);
  }, []);

  const renderMovies = (movies: IMovie[], loading: boolean) => {
    if (!loading) {
      if (movies.length) {
        return movies.map((item: IMovie) => {
          return (
            <MovieItem
              navigate={() => navigate(item.id)}
              {...item}
              key={item.id}
            ></MovieItem>
          );
        });
      } else {
        return <EmptyState>No films found</EmptyState>;
      }
    }
  };

  return (
    <ErrorBoundary>
      <AppWrapper>
        <HeaderContainer></HeaderContainer>

        <Switch>
          <Route exact path="/search">
            <SearchContainer></SearchContainer>
          </Route>

          <Route
            exact
            path="/film/:id"
            component={MovieDetailsContainer}
          ></Route>

          <Route path="*">
            <EmptyState>PAGE NOT FOUND</EmptyState>
            <Link to="/search">Go to Search</Link>
          </Route>
        </Switch>

        {history.location.pathname.includes('film') ||
        history.location.pathname.includes('search') ? (
          <div>
            <ListHeaderContainer
              listHeaderTitle={
                history.location.pathname.includes('search')
                  ? `${props.movies.length} movies found`
                  : `Movies by ${props.searchParams.searchBy}`
              }
            />
            <MoviesListWrapper id="movie-scroller" isEmptyList={false}>
              {renderMovies(props.movies, props.loading)}
            </MoviesListWrapper>
          </div>
        ) : null}
      </AppWrapper>
    </ErrorBoundary>
  );
};

const mapStateToProps = (state: any) => {
  return {
    movies: state.movies,
    loading: state.loading,
    error: state.error,
    selectedMovie: state.selectedMovie,
    searchParams: state.searchParams,
    route: state.route,
  };
};
const mapDispatchToProps = {
  loadMovies,
  getMovieById,
  setSearchParams,
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
