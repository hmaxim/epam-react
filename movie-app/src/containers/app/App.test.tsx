import 'jsdom-global/register';
import React, { Component } from 'react';
import { shallow, mount, render } from 'enzyme';
import 'jest-enzyme';
import 'jest-styled-components';
import App from './App';
import toJson from 'enzyme-to-json';

describe('App component', () => {
  let component: any;

  beforeEach(() => {
    component = render(<App />);
  });

  test('to be render', () => {  
    expect(toJson(component)).toMatchSnapshot();
  });
});
