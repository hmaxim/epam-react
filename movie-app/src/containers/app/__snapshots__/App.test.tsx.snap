// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component to be render 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background: #232323;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 20px;
}

.c2 {
  color: #f65261;
  height: 20px;
  font-size: 20px;
  padding: 20px 60px;
}

.c3 {
  color: #f65261;
  cursor: pointer;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 9.5%;
  color: white;
  background: #555555;
}

.c9 .search-buttons-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  padding: 20px 0;
}

.c9 .search-buttons-container span {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-right: 30px;
  font-size: 20px;
}

.c9 .search-buttons-container button {
  width: auto;
  border: none;
  cursor: pointer;
  height: 30px;
}

.c10 {
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c11 {
  margin-left: 65%;
}

.c7 {
  background: #F65261;
  border-radius: 3px;
  border-top-left-radius: 3px;
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 0px;
  border-top-right-radius: 0px;
  color: white;
  outline: none;
  text-transform: uppercase;
  width: auto;
  padding: 5px 10px;
}

.c8 {
  background: rgba(85,85,85,0.5);
  border-radius: 3px;
  border-top-left-radius: 0px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 3px;
  border-top-right-radius: 3px;
  color: white;
  outline: none;
  text-transform: uppercase;
  width: auto;
  padding: 5px 10px;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0 8%;
  overflow-y: auto;
  height: 59%;
}

.c13 {
  width: 323px;
  margin: 40px 20px;
}

.c13 .movie-info {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c13 .movie-info .movie-title {
  font-size: 20px;
  font-weight: 600;
}

.c13 .movie-info .movie-year {
  padding: 3px 5px;
  border: 1px solid white;
  border-radius: 3px;
}

.c4 {
  height: 25%;
  width: 100%;
}

.c4 .search-container {
  padding: 0 9.5%;
}

.c4 .search-container .search-title {
  color: #ffffff;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  margin-bottom: 10px;
  vertical-align: top;
}

.c4 .search-container button {
  width: 150px;
  height: 39.5px;
  font-size: 20px;
  vertical-align: top;
}

.c4 .search-container .search-buttons-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
  padding: 20px 0;
}

.c4 .search-container .search-buttons-container span {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  margin-right: 30px;
  font-size: 20px;
}

.c4 .search-container .search-buttons-container button {
  width: auto;
  border: none;
  cursor: pointer;
}

.c5 {
  width: calc(100% - 190px);
  height: 30px;
  padding: 5px 15px;
  color: #ffffff;
  margin-right: 10px;
  background: rgba(85,85,85,0.5);
  border: none;
  border-radius: 4px;
  outline: none;
}

.c5:focus {
  outline: none;
}

.c6 {
  background: #f65261;
  color: #ffffff;
  border-radius: 3px;
  border: none;
  cursor: pointer;
}

.c6:focus {
  outline: none;
}

.c6:active {
  opacity: 0.8;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      netflixRoulette
    </div>
    <div
      class="c3"
    />
  </div>
  <div
    class="c4"
  >
    <div
      class="search-container"
    >
      <h1
        class="search-title"
      >
        FIND YOUR MOVIE
      </h1>
      <div>
        <input
          class="c5"
        />
        <button
          class="c6"
        >
          SEARCH
        </button>
        <div
          class="search-buttons-container"
        >
          <span>
            SEARCH BY
          </span>
          <button
            class="c7"
            id="Title"
          >
            Title
          </button>
          <button
            class="c8"
            id="Genre"
          >
            Genre
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c9"
  >
    <div
      class="c10"
    >
       movie found
    </div>
    <div
      class="c11"
    >
      <div
        class="search-buttons-container"
      >
        <span>
          SORT BY
        </span>
        <button
          class="c7"
          id="Release Date"
        >
          Release Date
        </button>
        <button
          class="c8"
          id="Rating"
        >
          Rating
        </button>
      </div>
    </div>
  </div>
  <div
    class="c12"
    id="movie-scroller"
  >
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="movie-poster"
      >
        <img
          alt="poster"
          src="[object Object]"
        />
      </div>
      <div
        class="movie-info"
      >
        <div>
          <div
            class="movie-title"
          >
            Titanic
          </div>
          <span>
            Action
          </span>
        </div>
        <div
          class="movie-year"
        >
          1997
        </div>
      </div>
    </div>
  </div>
</div>
`;
